### PR TITLE
Improve subscription context serializer

### PR DESCRIPTION
**Pull request by stayallive**
- [x] Added or updated tests
- [ ] Added Docs for all relevant versions
- [x] Updated CHANGELOG.md

**Changes**

The current way the subscription serializer works is to serialize the `user` from the context and unserialize it on the other end. However this generates a lot of data to be stored and also could cause the user model to be outdated when unserialized.

This changes the subscription serializer to make use of the way Laravel's own queue component serializes models to serialize the context user and deserializes it on the other end.

I started working on this because I had some instances where the unserialization of the user model never worked (`unserialize(): Error at offset 11758 of 25737 bytes`) and prevented the subscription from even fireing.

In short:

- Less data stored in the subscription storage
- Always up-to-date user model in the context

**Breaking changes**

None.

### CODE REVIEW

File: src/Subscriptions/Serializer.php [hunk 1]
```diff
@@ -63,7 +66,16 @@ public function unserialize(string $context): GraphQLContext
 
         $request->setUserResolver(
             function () use ($rawUser) {
-                return unserialize($rawUser);
+                $user = $this->getRestoredPropertyValue($rawUser);
+
+                // This is here for backwards compatibility, before the Laravel
```
Comment (spawnia) [inline src/Subscriptions/Serializer.php:71]: To clarify, this is only needed so that current deployments can be upgraded seamlessly?

Comment (stayallive) [reply src/Subscriptions/Serializer.php:71]: Correct.

File: tests/Integration/Subscriptions/SerializerTest.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            $retrievedFromDatabase = true;
+        });
+
+        $unserialized = $serializer->unserialize(
+            $serializer->serialize($context)
+        );
+
+        $this->assertTrue($retrievedFromDatabase);
+        $this->assertInstanceOf(User::class, $unserialized->user());
+        $this->assertEquals($user->getKey(), $unserialized->user()->getKey());
```
Comment (spawnia) [inline tests/Integration/Subscriptions/SerializerTest.php:51]:
```suggestion
        $this->assertSame($user->getKey(), $unserialized->user()->getKey());
```

File: src/Subscriptions/Serializer.php (hunk 1 above)
Comment (spawnia) [reply src/Subscriptions/Serializer.php:71]: Cool, nicely done. Can you change the wording to reflect that this enables seamless upgrades to existing deployments and add a TODO to remove this in the next major version?

File: tests/Integration/Subscriptions/SerializerTest.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                'content' => '',
+            ],
+            'user' => serialize($context->user()),
+        ]);
+
+        $unserialized = $serializer->unserialize($serialized);
+
+        $this->assertFalse($retrievedFromDatabase);
+        $this->assertInstanceOf(User::class, $unserialized->user());
+        $this->assertEquals($user->getKey(), $unserialized->user()->getKey());
```
Comment (spawnia) [inline tests/Integration/Subscriptions/SerializerTest.php:94]:
```suggestion
        $this->assertSame($user->getKey(), $unserialized->user()->getKey());
```

**Comment by spawnia** [comment]
Great job, thank you :+1:
